tests/unit/util/test_file_converter.py

```diff
@@ -0,0 +1,157 @@
+"""Unit tests for functions in file_converter.py"""
+
+import os
+
+import pytest
+import pandas as pd
+import pathlib
+
+from clean_air.util import file_converter as fc
+
+
+@pytest.fixture()
+def excel_filepath(sampledir):
+    filepath = os.path.join(sampledir, "test_data",
+                            "metadata_form_responses.xlsx")
+    return filepath
+
+
+@pytest.fixture()
+def test_output_path(sampledir):
+    # Note: This path includes the first part of the filename (only omitting
+    # unit number).
+    test_output_path = os.path.join(sampledir, "test_data", "test_output")
+    return test_output_path
```

Comment:
maybe so.  will change this.  wasn't quite sure about it myself, changed it several times already.

Reply:
I'm no longer so sure about this one.  I don't like the test that was shown in the link; for a start it has 3 asserts in it, tests should only have one assert each, but that's beside the point really.  i might think about this a bit harder when my internet is working properly and i don't have so much pointless ridiculous admin to complete.  right now i can't see the value of making this change, but i'm  not dismissing it yet.

Reply:
Might have to talk about tests in general then because I though multiple asserts was fine...

I see a test as a function that does:
- set up some objects, possibly mock objects that don't actually do anything
- call a function/method
- check the result is as expected, which will probably need several asserts

In order to have only one assert per function, it would probably be best arranged in a class, where the setup and the function call happen in one method, and each assert appears in its own method. I *think* `unittest` is designed to be used like that. pytest does understand unittest tests, so we could do that, and I think we already did a bit in another test file, but everything else I've seen from pytest (which admittedly is not a lot) seems to prefer lumping several asserts in one function. And it does seem clearer to be fair.

Reply:
And actually the only change I was suggesting is more or less just replace the `sampledir` fixture with a `tmp_path` fixture, so that the output gets saved to a temporary directory instead of along with the sample data. During CI tests it won't matter because everything is done in a fresh checkout anyway, but when running the tests locally we'd end up with output data in our own checkout of `cap-sample-data` and may accidentally commit it.

Reply:
I don't necessarily disagree with you, but i do still need to pull files from cap-sample-data to test the conversions, so I'd still need to include ### sampledir.  But saving to a temporary path does seem more sensible , so I'll do a bit of cautious refactoring, see what i can do.

Reply:
alright, this was a tough change, but ultimately worthwhile.